### ISSUE TITLE
A Frida agent exception has occured error message - Objection Keychain dump

### ISSUE THREAD

**Issue by orelamihay86** [sensepost/objection#270]
Hi,
when using the keychain dump in iOS we received an error message that a Frida agent exception has occurred.
The Error type is: cannot read property 'bytes' of null

See the attached screenshot for the full error message and how to reproduce.

Expected behavior:
See a dump of the keychain.

![1](https://user-images.githubusercontent.com/29597895/64165630-21b82d00-ce4e-11e9-82d0-f22dee014122.png)

Best regards,
Orel

**Comment by leonjza** [comment]
Hey, thanks for the report! Could you invoke objection with the `--debug` flag so we can get a better stack trace to debug? Something like `objection --debug explore`.

**Comment by orelamihay86** [comment]
Sure! Thanks I will do it and let you know.

Have a nice day

**Comment by 0xABD** [comment]
The error I get is:
```
frida.core.RPCException: TypeError: undefined not callable (property 'keyEnumerator' of [object Object])
    at [anon] (../../../frida-gum/bindings/gumjs/duktape.c:65012)
    at src/ios/keychain.ts:192
    at src/ios/keychain.ts:111
    at map (native)
    at src/ios/keychain.ts:143
    at frida/runtime/message-dispatcher.js:15
    at c (frida/runtime/message-dispatcher.js:25)
```
It seems that either a `break` keyword is missing, or some checks need to be added:
https://github.com/sensepost/objection/blob/5ec821525865b61a7de822ddd6e0c8f0be7ab84e/agent/src/ios/keychain.ts#L187-L194


The issue seems very similar to https://github.com/chaitin/passionfruit/issues/58

**Comment by cakarlen** [comment]
I get the same error on my end. @0xABD

**Comment by leonjza** [comment]
@cakarlen got any steps to help me reproduce?

**Comment by cakarlen** [comment]
@leonjza I'm able to reproduce it by doing the following:
objection --debug --gadget [name] explore
ios keychain dump

This works on some apps right out of the box but I do know that this app, in particular, is written in Swift. I don't know if there is something different on the backend with ObjC/Swift but it seems to dump correctly on ObjC apps

**Comment by leonjza** [comment]
Sorry that question was silly of me, I meant to ask if you have an application you can share with me that I can reproduce this with. Even if via say a DM on Twitter (I use the same handle there).

Thanks!

**Comment by cakarlen** [comment]
Sent you a DM via Twitter. Thanks for the help

**Comment by leonjza** [comment]
Managed to replicate this locally. Thanks!

**Comment by leonjza** [comment]
Hopefully fixed in https://github.com/sensepost/objection/commit/8ed43663a5cf29c8355e393935b628f8e471cdb2. Will land in next tag.

**Comment by leonjza** [comment]
Should be fixed now in 1.7.5.
